README.md

```diff
@@ -1,2 +1,187 @@
-# py-composite-action-lib
-PyCompositeActionLib is a Python library that streamlines creating and managing composite GitHub Actions. It offers reusable classes and scripts for controlling action inputs/outputs and accessing the GitHub API. 
+# PyCompositeActionLib
+
+- [Description](#description)
+- [Motivation](#motivation)
+- [How to Use](#how-to-use)
+  - [Example: Control Action Inputs and Outputs](#example-control-action-inputs-and-outputs)
+  - [Example: Access GitHub API with GithubManager](#example-access-github-api-with-githubmanager)
+- [Features](#features)
+- [How to Use Local Version of Library for Development](#how-to-use-local-version-of-library-for-development)
+- [How to Run Tests](#how-to-run-tests)
+- [How to Release](#how-to-release)
+- [Contribution Guidelines](#contribution-guidelines)
+- [License Information](#license-information)
+- [Contact or Support Information](#contact-or-support-information)
+
+
+## Description
+**PyCompositeActionLib** is a Python library that streamlines creating and managing composite GitHub Actions. It offers
+reusable classes and scripts for controlling action inputs/outputs and accessing the GitHub API. 
+
+## Motivation
+The motivation behind PyCompositeActionLib is to provide developers with a robust set of tools that simplify
+the creation of composite GitHub Actions. By leveraging reusable components and streamlined API access, this library
+aims to reduce redundancy, improve efficiency, and enhance the overall development experience for creating complex
+workflows on GitHub.
+
+## How to use
+### Example: Control Action Inputs and Outputs
+```python
+from py_composite_action_lib import get_action_input, set_action_output, set_action_failed
+
+# Example: Get action input
+repo_name = get_action_input('repo-name')
+if not repo_name:
+    set_action_failed('The input "repo-name" is required.')
+
+# Example: Set action output
+set_action_output('output-name', 'success')
+```
+
+### Example: Access GitHub API with GithubManager
+```python
+from github import Github, Auth
+from py_composite_action_lib import GithubManager
+
+# Authenticate to GitHub
+token = 'your_github_token'
```

Comment:
please provide a better example of handling secrets first

Reply:
Addressed in commit 58ba9758ebbc98cc876c7e5df2285f9a9f20c407.
The action is accessing inputs from env variables.

See the usage of action which is using this library:

```python
- name: Generate Release Notes
  id: release_notes_generator
  uses: AbsaOSS/generate-release-notes@v0.2.0
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  
  with:
    tag-name: "v0.1.0"
    chapters: '[
      {"title": "Breaking Changes 💥", "label": "breaking-change"},
      {"title": "New Features 🎉", "label": "enhancement"},
      {"title": "New Features 🎉", "label": "feature"},
      {"title": "Bugfixes 🛠", "label": "bug"}
    ]'
    warnings: false
    published-at: true
    skip-release-notes-label: 'ignore-in-release'     # changing default value of label
    print-empty-chapters: false
    chapters-to-pr-without-issue: false
    verbose: false
```